core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -113,6 +113,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   /**
    * :: DeveloperApi ::
    * Alternative constructor for setting preferred locations where Spark will create executors.
+   * Note that preferred locations feature does not work as it is supposed to, see SPARK-8949
```

Comment:
I would be more explicit and say that passing in preferred node location data has no effect at all.